### ISSUE TITLE
virtual/blas is satisfied by dev-cpp/eigen, but eigen doesn't provide blas libraries

### ISSUE THREAD

**Issue by rayhogenson** [gentoo/sci#793]
I have installed virtual/blas and dev-cpp/eigen, satisfying all of the requirements for these packages, but now some packages, for example sci-libs/umfpack won't build, citing the error "configure: error: Cannot find blas libraries".

It does not appear to me that eigen is capable of providing blas, and unless I'm misunderstanding the situation (in which case can someone help me to figure out why these packages won't build), I believe eigen should be removed from the list of dependencies for virtual/blas so that it properly installs something like blas-reference in its place.

**Comment by jlec** [comment]
Are you installing eigen from the overlay or main tree?

**Comment by rayhogenson** [comment]
Both versions exhibit the same behavior.

**Comment by kiwifb** [comment]
Question: Have you enabled the useflag "fortran" for eigen? That could be a mistake in the virtual since blas is a fortran library I am expecting the useflag to be required. However the flag is not present in the dependency in `virtual/blas`.

**Comment by rayhogenson** [comment]
Emerging eigen 3.2.8-r2 from the overlay with the use flag fortran (the newer eigen 3.3.3 from the main tree has no such use flag) changes nothing, the same packages fail to build.

**Comment by kiwifb** [comment]
Can you paste the log or post it somewhere?

**Comment by rayhogenson** [comment]
I assume you mean the build log, here it is:
```
 * Package:    sci-libs/umfpack-5.6.2
 * Repository: gentoo
 * Maintainer: sci@gentoo.org
 * USE:        abi_x86_64 amd64 cholmod elibc_glibc kernel_linux userland_GNU
 * FEATURES:   compressdebug preserve-libs sandbox splitdebug userpriv usersandbox
>>> Unpacking source...
>>> Unpacking umfpack-5.6.2.tar.bz2 to /var/tmp/portage/sci-libs/umfpack-5.6.2/work
>>> Source unpacked in /var/tmp/portage/sci-libs/umfpack-5.6.2/work
>>> Preparing source in /var/tmp/portage/sci-libs/umfpack-5.6.2/work/umfpack-5.6.2 ...
 * Running elibtoolize in: umfpack-5.6.2/
 *   Applying target-nm/2.4.2 patch ...
 *   Applying ppc64le/2.4.2 patch ...
 * Running elibtoolize in: umfpack-5.6.2/config/
 *   Applying portage/1.2.0 patch ...
 *   Applying sed/1.5.6 patch ...
 *   Applying as-needed/2.4.2 patch ...
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/sci-libs/umfpack-5.6.2/work/umfpack-5.6.2 ...
Package blas was not found in the pkg-config search path.
Perhaps you should add the directory containing `blas.pc'
to the PKG_CONFIG_PATH environment variable
No package 'blas' found
>>> Working in BUILD_DIR: "/var/tmp/portage/sci-libs/umfpack-5.6.2/work/umfpack-5.6.2_build"
 * econf: updating umfpack-5.6.2/config/config.guess with /usr/share/gnuconfig/config.guess
 * econf: updating umfpack-5.6.2/config/config.sub with /usr/share/gnuconfig/config.sub
/var/tmp/portage/sci-libs/umfpack-5.6.2/work/umfpack-5.6.2/configure --prefix=/usr --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --sysconfdir=/etc --localstatedir=/var/lib --disable-dependency-tracking --disable-silent-rules --libdir=/usr/lib64 --docdir=/usr/share/doc/umfpack-5.6.2 --enable-shared --disable-static --with-blas= --without-doc --with-cholmod
checking for a BSD-compatible install... /usr/lib/portage/python3.4/ebuild-helpers/xattr/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for style of include used by make... GNU
checking for x86_64-pc-linux-gnu-gcc... x86_64-pc-linux-gnu-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether x86_64-pc-linux-gnu-gcc accepts -g... yes
checking for x86_64-pc-linux-gnu-gcc option to accept ISO C89... none needed
checking dependency style of x86_64-pc-linux-gnu-gcc... none
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for ld used by x86_64-pc-linux-gnu-gcc... /usr/x86_64-pc-linux-gnu/bin/ld
checking if the linker (/usr/x86_64-pc-linux-gnu/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/x86_64-pc-linux-gnu-nm -B
checking the name lister (/usr/bin/x86_64-pc-linux-gnu-nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/x86_64-pc-linux-gnu/bin/ld option to reload object files... -r
checking for x86_64-pc-linux-gnu-objdump... x86_64-pc-linux-gnu-objdump
checking how to recognize dependent libraries... pass_all
checking for x86_64-pc-linux-gnu-dlltool... no
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for x86_64-pc-linux-gnu-ar... x86_64-pc-linux-gnu-ar
checking for archiver @FILE support... @
checking for x86_64-pc-linux-gnu-strip... x86_64-pc-linux-gnu-strip
checking for x86_64-pc-linux-gnu-ranlib... x86_64-pc-linux-gnu-ranlib
checking command to parse /usr/bin/x86_64-pc-linux-gnu-nm -B output from x86_64-pc-linux-gnu-gcc object... ok
checking for sysroot... no
checking for x86_64-pc-linux-gnu-mt... no
checking for mt... no
checking if : is a manifest tool... no
checking how to run the C preprocessor... x86_64-pc-linux-gnu-gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if x86_64-pc-linux-gnu-gcc supports -fno-rtti -fno-exceptions... no
checking for x86_64-pc-linux-gnu-gcc option to produce PIC... -fPIC -DPIC
checking if x86_64-pc-linux-gnu-gcc PIC flag -fPIC -DPIC works... yes
checking if x86_64-pc-linux-gnu-gcc static flag -static works... yes
checking if x86_64-pc-linux-gnu-gcc supports -c -o file.o... yes
checking if x86_64-pc-linux-gnu-gcc supports -c -o file.o... (cached) yes
checking whether the x86_64-pc-linux-gnu-gcc linker (/usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for x86_64-pc-linux-gnu-gcc... (cached) x86_64-pc-linux-gnu-gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether x86_64-pc-linux-gnu-gcc accepts -g... (cached) yes
checking for x86_64-pc-linux-gnu-gcc option to accept ISO C89... (cached) none needed
checking dependency style of x86_64-pc-linux-gnu-gcc... (cached) none
checking float.h usability... yes
checking float.h presence... yes
checking for float.h... yes
checking for string.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for size_t... yes
checking for error_at_line... yes
checking for stdlib.h... (cached) yes
checking for GNU libc compatible malloc... yes
checking for pow in -lm... yes
checking whether to build user guide... no
checking for x86_64-pc-linux-gnu-pkg-config... /usr/bin/x86_64-pc-linux-gnu-pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for SUITESPARSECONFIG... yes
checking SuiteSparse_config.h usability... yes
checking SuiteSparse_config.h presence... yes
checking for SuiteSparse_config.h... yes
checking for SuiteSparse_malloc in -lsuitesparseconfig... yes
checking for AMD... yes
checking amd.h usability... yes
checking amd.h presence... yes
checking for amd.h... yes
checking for amd_aat in -lamd... yes
checking for x86_64-pc-linux-gnu-g77... no
checking for x86_64-pc-linux-gnu-xlf... no
checking for x86_64-pc-linux-gnu-f77... no
checking for x86_64-pc-linux-gnu-frt... no
checking for x86_64-pc-linux-gnu-pgf77... no
checking for x86_64-pc-linux-gnu-cf77... no
checking for x86_64-pc-linux-gnu-fort77... no
checking for x86_64-pc-linux-gnu-fl32... no
checking for x86_64-pc-linux-gnu-af77... no
checking for x86_64-pc-linux-gnu-xlf90... no
checking for x86_64-pc-linux-gnu-f90... no
checking for x86_64-pc-linux-gnu-pgf90... no
checking for x86_64-pc-linux-gnu-pghpf... no
checking for x86_64-pc-linux-gnu-epcf90... no
checking for x86_64-pc-linux-gnu-gfortran... x86_64-pc-linux-gnu-gfortran
checking whether we are using the GNU Fortran 77 compiler... yes
checking whether x86_64-pc-linux-gnu-gfortran accepts -g... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for x86_64-pc-linux-gnu-gfortran option to produce PIC... -fPIC
checking if x86_64-pc-linux-gnu-gfortran PIC flag -fPIC works... yes
checking if x86_64-pc-linux-gnu-gfortran static flag -static works... yes
checking if x86_64-pc-linux-gnu-gfortran supports -c -o file.o... yes
checking if x86_64-pc-linux-gnu-gfortran supports -c -o file.o... (cached) yes
checking whether the x86_64-pc-linux-gnu-gfortran linker (/usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking how to get verbose linking output from x86_64-pc-linux-gnu-gfortran... -v
checking for Fortran 77 libraries of x86_64-pc-linux-gnu-gfortran...  -L/usr/lib/gcc/x86_64-pc-linux-gnu/6.3.0 -L/usr/lib/gcc/x86_64-pc-linux-gnu/6.3.0/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/usr/lib/gcc/x86_64-pc-linux-gnu/6.3.0/../../../../x86_64-pc-linux-gnu/lib -L/usr/lib/gcc/x86_64-pc-linux-gnu/6.3.0/../../.. -lgfortran -lm -lquadmath
checking for dummy main to link with Fortran 77 libraries... none
checking for Fortran 77 name-mangling scheme... lower case, underscore, no extra underscore
checking for BLAS... no
checking if sgemm_ is being linked in already... no
checking for sgemm_ in -lopenblas... no
checking for ATL_xerbla in -latlas... no
checking for sgemm_ in -lblas... no
checking for sgemm_ in -lmkl_intel_lp64... no
checking for sgemm_ in -lmkl... no
checking for sgemm_ in -framework vecLib... no
checking for sgemm_ in -lcxml... no
checking for sgemm_ in -ldxml... no
checking for sgemm_ in -lscs... no
checking for sgemm_ in -lcomplib.sgimath... no
checking for sgemm_ in -lblas... (cached) no
checking for sgemm_ in -lblas... (cached) no
configure: error: Cannot find blas libraries

!!! Please attach the following file when seeking support:
!!! /var/tmp/portage/sci-libs/umfpack-5.6.2/work/umfpack-5.6.2_build/config.log
 * ERROR: sci-libs/umfpack-5.6.2::gentoo failed (configure phase):
 *   econf failed
 * 
 * Call stack:
 *               ebuild.sh, line  115:  Called src_configure
 *             environment, line 2771:  Called autotools-utils_src_configure
 *             environment, line  538:  Called econf '--docdir=/usr/share/doc/umfpack-5.6.2' '--enable-shared' '--disable-static' '--with-blas=' '--without-doc' '--with-cholmod'
 *        phase-helpers.sh, line  665:  Called __helpers_die 'econf failed'
 *   isolated-functions.sh, line  117:  Called die
 * The specific snippet of code:
 *   		die "$@"
 * 
 * If you need support, post the output of `emerge --info '=sci-libs/umfpack-5.6.2::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=sci-libs/umfpack-5.6.2::gentoo'`.
 * The complete build log is located at '/var/tmp/portage/sci-libs/umfpack-5.6.2/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/sci-libs/umfpack-5.6.2/temp/environment'.
 * Working directory: '/var/tmp/portage/sci-libs/umfpack-5.6.2/work/umfpack-5.6.2_build'
 * S: '/var/tmp/portage/sci-libs/umfpack-5.6.2/work/umfpack-5.6.2'
```

**Comment by kiwifb** [comment]
It appears that `blas.pc` is missing in action. What does `eselect blas list` reports?

**Comment by rayhogenson** [comment]
```
Available providers for blas:
  (none found)
```

**Comment by kiwifb** [comment]
You have eigen from the science overlay?

**Comment by kiwifb** [comment]
Furthermore do you have a file `/usr/lib64/pkgconfig/eigen_blas.pc` installed?

**Comment by rayhogenson** [comment]
I have installed eigen-3.2.8-r2 from the science overlay, but no file `/usr/lib64/pkgconfig/eigen_blas.pc`.

There is, however, a file `/usr/lib64/pkgconfig/blas.pc`, a broken symbolic link to `../../../etc/env.d/alternatives/blas/_current/usr/lib64/pkgconfig/blas.pc`, I'm guessing this file is left over from when I had blas-reference installed.

**Comment by rayhogenson** [comment]
I have installed eigen-3.2.8-r2 from the science overlay, but no file `/usr/lib64/pkgconfig/eigen_blas.pc`.

There is, however, a file `/usr/lib64/pkgconfig/blas.pc`, a broken symbolic link to `../../../etc/env.d/alternatives/blas/_current/usr/lib64/pkgconfig/blas.pc`, I'm guessing this file is left over from when I had blas-reference installed.

**Comment by kiwifb** [comment]
When I install here, with the fortran useflag, I get the `eigen_blas.pc` file. Output of `eix -I eigen` please.

**Comment by rayhogenson** [comment]
Ah, it appears that I am mistaken. I didn't have eigen from the science overlay installed. After installing the package from the overlay, the issues are fixed. I feel stupid for missing that.

One issue remains, portage wants to upgrade eigen to the newer version present in the main tree, but for now I can mask the newer versions in order to prevent this.

Thanks for your help.

**Comment by kiwifb** [comment]
OK, now someone with power over the overlay must make sure that `virtual/blas` depends on `dev-cpp/eigen[fortran]`.

**Comment by devurandom** [comment]
Can this be closed, since #795 was merged?

**Comment by kiwifb** [comment]
I'd say yes.
